File: Examples/CSharp/ModifyingAndConvertingImages/DitheringRasterImages.cs
// GIST-ID: 2890e005f3d8b07752bc3ceb222d7cc8
using Aspose.Imaging.FileFormats.Jpeg;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class DitheringRasterImages
    {
        public static void Run()
        {
            Console.WriteLine("Running example DitheringRasterImages");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Create an instance of JpegImage and load an image as a JpegImage.
            using (var image = (JpegImage)Image.Load(dataDir + "aspose-logo.jpg"))
            {
                // Perform Floyd‑Steinberg dithering on the current image and save the resultant image.
                image.Dither(DitheringMethod.ThresholdDithering, 4);
                image.Save(dataDir + "SampleImage_out.bmp");
            }

            Console.WriteLine("Finished example DitheringRasterImages");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/DitheringRasterImages.vb
Imports Aspose.Imaging

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
' If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class DitheringRasterImages
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")

            ' ExStart:DitheringRasterImages
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Create an instance of JpegImage and load an image as of JpegImage
            Using image__1 = DirectCast(Image.Load(dataDir & Convert.ToString("aspose-logo.jpg")), FileFormats.Jpeg.JpegImage)
                ' Peform Floyd Steinberg dithering on the current image
                image__1.Dither(DitheringMethod.ThresholdDithering, 4)
                ' Save the resultant image
                image__1.Save(dataDir & Convert.ToString("SampleImage_out.bmp"))
            End Using
            ' ExEnd:DitheringRasterImages
        End Sub
    End Class
End Namespace
